src/Configuration/Config/src/ConfigurationKeyComparer.cs

```diff
@@ -26,8 +26,8 @@ public class ConfigurationKeyComparer : IComparer<string>
         /// <returns></returns>
         public int Compare(string x, string y)
         {
-            var xParts = x?.Split(_keyDelimiterArray, StringSplitOptions.RemoveEmptyEntries) ?? new string[0];
-            var yParts = y?.Split(_keyDelimiterArray, StringSplitOptions.RemoveEmptyEntries) ?? new string[0];
+            var xParts = x?.Split(_keyDelimiterArray, StringSplitOptions.RemoveEmptyEntries) ?? Array.Empty<string>();
+            var yParts = y?.Split(_keyDelimiterArray, StringSplitOptions.RemoveEmptyEntries) ?? Array.Empty<string>();
 
             // Compare each part until we get two parts that are not equal
             for (int i = 0; i < Math.Min(xParts.Length, yParts.Length); i++)
```

Comment:
```suggestion
            var n = Math.Min(xParts.Length, yParts.Length);
            for (var i = 0; i < n; i++)
```
Otherwise `n` is evaluated at each iteration.